patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1113,6 +1145,30 @@
  			else if ((double)player[myPlayer].position.Y >= worldSurface * 16.0 + (double)(screenHeight / 2) && !WorldGen.oceanDepths((int)(screenPosition.X + (float)(screenWidth / 2)) / 16, (int)(screenPosition.Y + (float)(screenHeight / 2)) / 16)) {
  				if (player[myPlayer].ZoneHallow) {
  					newMusic = 11;
+@@ -9875,12 +_,12 @@
+ 			float num3 = Math.Max(1f, 1f + 4f * cloudAlpha);
+ 			if (cloudBGActive > 0f) {
+ 				if (cloudBGActive > 1f)
+-					cloudBGActive -= (float)dayRate / num3;
++					cloudBGActive -= (float)desiredWorldEventsUpdateRate / num3;
+ 
+ 				if (cloudBGActive < 1f)
+ 					cloudBGActive = 1f;
+ 
+-				if (cloudBGActive == 1f && rand.Next((int)((float)(num2 * 2 / Math.Max(dayRate, 1)) * num3)) == 0) {
++				if (cloudBGActive == 1f && rand.Next((int)((float)(num2 * 2 / Math.Max(desiredWorldEventsUpdateRate, 1)) * num3)) == 0) {
```

Comment:
These int casts seem like they'll round to 0 in some cases. Should use a float based rand.Next